lib/web/apiserver.go

```diff
@@ -1772,6 +1777,46 @@ func (h *Handler) changeUserAuthentication(w http.ResponseWriter, r *http.Reques
 	}, nil
 }
 
+func (h *Handler) trySettingConnectorNameToPasswordless(ctx *SessionContext, r *http.Request, req changeUserAuthenticationRequest) error {
+	if req.WebauthnCreationResponse == nil {
+		return nil
+	}
+
+	if !h.ClusterFeatures.GetCloud() {
+		return nil
+	}
+
+	authPreference, err := ctx.clt.GetAuthPreference(r.Context())
+
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if authPreference.GetConnectorName() == constants.PasswordlessConnector {
+		return nil
+	}
+
+	users, err := h.cfg.ProxyClient.GetUsers(false)
```

Comment:
do you know if this call uses the proxy's cache of users?

Reply:
Yeah, users are pulled from cache